### ISSUE TITLE
dateWidget template is not the same over time

### ISSUE THREAD

**Issue by rauul338** [Holt59/cakephp3-bootstrap3-helpers#35]
I have a Reservations model that require a start date and end date.
Using the FormHelper in the simplest way
```php
<fieldset>
    <legend><?= __('Request reservation') ?></legend>
<?php
    echo $this->Form->input('date_start', ['type' => 'date']);
    echo $this->Form->input('date_end', ['type' => 'date']);
?>
</fieldset>
```
gives this result
![1](https://cloud.githubusercontent.com/assets/7199046/8512884/f5198efe-232c-11e5-968c-c3d6d4eedc90.png)

Adding some debug in the [date function](https://github.com/Holt59/cakephp3-bootstrap3-helpers/blob/master/src/View/Helper/BootstrapFormHelper.php#L554) returned this
![2](https://cloud.githubusercontent.com/assets/7199046/8512896/3a40e478-232d-11e5-8629-6ccab789408b.png)

If I reset the previous template after the call to parent::date it works :)

```php
public function date($fieldName, array $options = []) {
    $fields = ['year' => true, 'month' => true, 'day' => true];
    $prev_template = $this->templates('dateWidget');
    // debug('dateWidget: before '.$this->templates('dateWidget'));
    $this->templates([
        'dateWidget' => $this->_getDatetimeTemplate($fields, $options)
    ]);
    // debug('dateWidget: after '.$this->templates('dateWidget'));
    $return = parent::date($fieldName, $options);
    $this->templates(['dateWidget' => $prev_template]);
    return $return;
}
```
I hope I helped. It's my first time reporting an issue :+1:

**Comment by Holt59** [comment]
Should be fixed with commit 1f693960d28e134b9ef4188a2fb869564abce44f. Thanks for targeting the bug so precisely ;) I did not directly use your code since I had to put it in the three methods (`date`, `datetime` and `time`) but the idea is all yours!

**Comment by rauul338** [comment]
Thamks!
It's fixed now. The code was only a draft. :smile: 

Thanks again for such awesome plugin!
